src/apps/cardano/sign_tx.py

```diff
@@ -214,32 +192,30 @@ def _build_witnesses(self, tx_aux_hash: str):
             )
             witnesses.append(
                 [
-                    self.types[index],
+                    (input.type or 0),
```

Comment:
why `input.type or 0`? same below

Reply:
See https://github.com/trezor/trezor-core/blob/master/src/apps/cardano/sign_tx.py , I just rewrote the code without the `types` list.

Reply:
hmm interesting. I would rather drop the `or 0` then. I looked at the protobuf definition and it defaults to `0` anyway

Reply:
ok, perhaps let's keep it there for the sake of readability: https://github.com/trezor/connect/blob/c7fb0032a8475cc09ec29ccc860fbe76066f4169/src/js/types/trezor.js#L601 in js the type field seems is optional so one might wonder to what it does default and this line of code would make that clear with the `or 0`. I've found that pattern also in other parts of the codebase